### ISSUE TITLE
Multi-target test run KeyError with differing test cases

### ISSUE THREAD

**Issue by bridadan** [ARMmbed/greentea#186]
If you do a multi-target test run (ex `mbedgt -V -v -t K64F-GCC_ARM,NUCLEO_F401RE-GCC_ARM --report-html=report.html`) and attempt to save an HTML report, you get the following traceback:

```
mbedgt: all tests finished!
mbedgt: shuffle seed: 0.9936295739
mbedgt: exporting to HTML file 'report.html'...
mbedgt: unexpected error:
        u'features-net-feature_ipv4-tests-mbedmicro-net-udp_echo_client'
Traceback (most recent call last):
  File "C:\Python27\Scripts\mbedgt-script.py", line 9, in <module>
    load_entry_point('mbed-greentea', 'console_scripts', 'mbedgt')()
  File "c:\users\bridan01\documents\dev\greentea\mbed_greentea\mbed_greentea_cli.py", line 389, in main
    cli_ret = main_cli(opts, args)
  File "c:\users\bridan01\documents\dev\greentea\mbed_greentea\mbed_greentea_cli.py", line 1007, in main_cli
    html_report = exporter_html(test_report)
  File "c:\users\bridan01\documents\dev\greentea\mbed_greentea\mbed_report_api.py", line 646, in exporter_html
    test_results = test_result_ext["%s-%s" % (platform, toolchain)][test_name]
KeyError: u'features-net-feature_ipv4-tests-mbedmicro-net-udp_echo_client'
```

**Comment by ciarmcom** [comment]
ARM Internal Ref: IOTSFW-2899

**Comment by bridadan** [comment]
Should be fixed with #190
